ambry-quota/src/main/java/com/github/ambry/quota/PostProcessQuotaChargeCallback.java

```diff
@@ -24,30 +24,68 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(QuotaChargeCallback.class);
   private final QuotaManager quotaManager;
   private final RestRequest restRequest;
   private final RequestQuotaCostPolicy requestCostPolicy;
   private final boolean isQuotaEnforcedOnRequest;
 
   /**
-   * Constructor for {@link RejectingQuotaChargeCallback}.
+   * Constructor for {@link PostProcessQuotaChargeCallback}.
    * @param quotaManager {@link QuotaManager} object responsible for charging the quota.
    * @param restRequest {@link RestRequest} for which quota is being charged.
    * @param isQuotaEnforcedOnRequest flag indicating if request quota should be enforced after charging. Requests like
    *                                 update ttl, delete etc are charged, but quota is not enforced on them.
    */
-  public RejectingQuotaChargeCallback(QuotaManager quotaManager, RestRequest restRequest,
+  public PostProcessQuotaChargeCallback(QuotaManager quotaManager, RestRequest restRequest,
       boolean isQuotaEnforcedOnRequest) {
     this.quotaManager = quotaManager;
     requestCostPolicy = new SimpleRequestQuotaCostPolicy(quotaManager.getQuotaConfig());
     this.restRequest = restRequest;
     this.isQuotaEnforcedOnRequest = isQuotaEnforcedOnRequest;
   }
 
+  @Override
+  public QuotaAction checkAndCharge(boolean shouldCheckQuotaExceedAllowed, boolean forceCharge, long chunkSize) throws QuotaException {
+    QuotaAction quotaAction = QuotaAction.ALLOW;
+    try {
+      Map<QuotaName, Double> requestCost = requestCostPolicy.calculateRequestQuotaCharge(restRequest, chunkSize)
+          .entrySet()
+          .stream()
+          .collect(Collectors.toMap(entry -> QuotaName.valueOf(entry.getKey()), Map.Entry::getValue));
+      quotaAction = quotaManager.chargeAndRecommend(restRequest, requestCost, false, true);
+      if (QuotaUtils.shouldThrottle(quotaAction) && isQuotaEnforcedOnRequest) {
+        if (quotaManager.getQuotaMode() == QuotaMode.THROTTLING
+            && quotaManager.getQuotaConfig().throttleInProgressRequests) {
+          throw new QuotaException("Exception while charging quota",
+              new RouterException("RequestQuotaExceeded", RouterErrorCode.TooManyRequests), false);
+        } else {
+          LOGGER.debug("Quota exceeded for an in progress request.");
+        }
+      }
+    } catch (Exception ex) {
+      if (ex.getCause() instanceof RouterException && ((RouterException) ex.getCause()).getErrorCode()
+          .equals(RouterErrorCode.TooManyRequests)) {
+        throw ex;
+      }
+      LOGGER.error("Unexpected exception while charging quota.", ex);
+    }
+    return quotaAction;
+  }
+
+  @Override
+  public QuotaAction checkAndCharge(boolean shouldCheckQuotaExceedAllowed, boolean forceCharge) throws QuotaException {
+    return checkAndCharge(shouldCheckQuotaExceedAllowed, forceCharge, quotaManager.getQuotaConfig().quotaAccountingUnit);
+  }
```

Comment:
we don't need this implementation if it's not different from the default one.

Reply:
removed.